File: task3/php/main.php
<?php
$a = (int)readline("Введите количество чисел: ");  // Получаем количество чисел для обработки

for($n = 0; $n < $a; $n++) {
   $num = (int)readline("Введите число: ");  // Читаем очередное число
   $digSum = 0;  // Инициализация суммы цифр (исправлено с 1 на 0)
   $digWork = 1;  // Инициализация произведения цифр
    
    while ($num > 0) {
        $dig = $num % 10;  // Получаем последнюю цифру
        $num = (int)($num / 10);  // Удаляем последнюю цифру
        $digSum += $dig;  // Добавляем цифру к сумме
        $digWork *= $dig;  // Умножаем на произведение
    }
    
    if ($digSum < $digWork) {  // Проверяем условие
        echo $n . "\n";  // Выводим индекс числа (не само число!)
    }
}
?>

File: task3/ruby/main.rb
print "Введите количество чисел: "
input = gets.to_i  # Чтение количества чисел 

(0...input).each do |n|  # Цикл по диапазону
  print "Введите число: "
  num = gets.to_i  # Чтение числа
  
  dig_sum = 0  # Сумма цифр
  dig_work = 1  # Произведение цифр
  
  while num > 0  # Разбор числа на цифры
    dig = num % 10  # Получение последней цифры
    num /= 10  # Удаление обработанной цифры
    dig_sum += dig  # Накопление суммы
    dig_work *= dig  # Накопление произведения
  end

  puts n if dig_sum < dig_work  # Условный вывод в одну строку
end